implementations/rust/ockam/ockam_command/src/util/output.rs

```diff
@@ -0,0 +1,48 @@
+use cli_table::{Cell, Style, Table};
+use core::fmt::Write;
+
+use ockam_api::cloud::space::Space;
+use ockam_core::CowStr;
+
+pub trait Output {
+    fn output(&self) -> anyhow::Result<String>;
+}
+
+impl Output for Space<'_> {
+    fn output(&self) -> anyhow::Result<String> {
+        let mut w = String::new();
+        write!(w, "id: {}", self.id)?;
+        write!(w, "\nname: {}", self.name)?;
+        write!(w, "\nusers: {}", CowStr::slice_to_string(&self.users))?;
+        Ok(w)
+    }
+}
+
+impl Output for Vec<Space<'_>> {
+    fn output(&self) -> anyhow::Result<String> {
```

Comment:
Stylistically I find a for loop like `for Space { id, name, users, .. } in self { ... }` easier to read compared to a fold, but YMMV.